src/coreclr/vm/eventing/eventpipe/ep-rt-coreclr.h

```diff
@@ -2030,6 +2030,7 @@ ep_rt_thread_create (
 				DWORD thread_id = 0;
 				HANDLE server_thread = ::CreateThread (nullptr, 0, reinterpret_cast<LPTHREAD_START_ROUTINE>(thread_func), nullptr, 0, &thread_id);
 				if (server_thread != NULL) {
+					::SetThreadName(server_thread, W(".NET EventPipe"));
```

Comment:
Same here

Reply:
Could you help me on this?

I need proper way to use pthread APIs in `src/native/eventpipe/ds-server.c`
When I add compile options like below in `src/coreclr/vm/eventing/eventpipe/CMakeLists.txt`
```cmake
find_package(Threads REQUIRED)

add_library_clr(eventpipe_objs OBJECT ${EVENTPIPE_SOURCES})
target_precompile_headers(eventpipe_objs PRIVATE [["common.h"]])
set_target_properties(eventpipe_objs PROPERTIES LINKER_LANGUAGE CXX)
add_dependencies(eventpipe_objs eventing_headers)

set_property(TARGET eventpipe_objs PROPERTY COMPILE_OPTIONS "-pthread")
set_property(TARGET eventpipe_objs PROPERTY INTERFACE_COMPILE_OPTIONS "-pthread")
target_link_libraries(eventpipe_objs "${CMAKE_THREAD_LIBS_INIT}")
```

Build error occurs like this
```
  In file included from <built-in>:1:
  In file included from /home/leee/dotnet/runtime/artifacts/obj/coreclr/Linux.x64.Debug/vm/eventing/eventpipe/CMakeFiles/eventpipe_objs.dir/cmake_pch.hxx:5:
  In file included from /home/leee/dotnet/runtime/src/coreclr/vm/common.h:94:
  In file included from /home/leee/dotnet/runtime/src/coreclr/vm/util.hpp:15:
  In file included from /home/leee/dotnet/runtime/src/coreclr/inc/metadata.h:16:
  In file included from /home/leee/dotnet/runtime/src/coreclr/inc/ex.h:18:
  In file included from /home/leee/dotnet/runtime/src/coreclr/inc/sstring.h:46:
  In file included from /home/leee/dotnet/runtime/src/coreclr/inc/sbuffer.h:38:
  /home/leee/dotnet/runtime/src/coreclr/inc/iterator.h:222:43: error: expected ';' after expression
          __if_exists(CONTAINER::m_revision)
                                            ^
  /home/leee/dotnet/runtime/src/coreclr/inc/iterator.h:245:43: error: expected ';' after expression
          __if_exists(CONTAINER::m_revision)
                                            ^
  /home/leee/dotnet/runtime/src/coreclr/inc/iterator.h:258:43: error: expected ';' after expression
          __if_exists(CONTAINER::m_revision)
                                            ^
```

Newer methods like below is not working
```
set(THREADS_PREFER_PTHREAD_FLAG ON)
find_package(Threads REQUIRED)
target_link_libraries(eventpipe_objs PRIVATE Threads::Threads)
```

Reply:
You cannot use OS headers in CoreCLR. Everything has to go through PAL currently.

Reply:
Thank you for the advice. Fixed.